react/features/app/actions.js

```diff
@@ -31,176 +31,70 @@ declare var APP: Object;
-    // navigating to the WelcomePage - the perceived/visible navigation will be
-    // faster.
-    if (!room && typeof APP === 'undefined') {
-        return Promise.resolve();
-    }
+        location.protocol || (location.protocol = 'https:');
+        const locationURL = new URL(location.toString());
 
-    /* eslint-disable no-param-reassign */
+        dispatch(configWillLoad(locationURL));
 
-    protocol = protocol.toLowerCase();
+        let protocol = location.protocol.toLowerCase();
+        const { contextRoot, host, room } = location;
 
-    // The React Native app supports an app-specific scheme which is sure to not
-    // be supported by fetch (or whatever loadConfig utilizes).
-    protocol !== 'http:' && protocol !== 'https:' && (protocol = 'https:');
+        // The React Native app supports an app-specific scheme which is sure to not
+        // be supported by fetch (or whatever loadConfig utilizes).
+        protocol !== 'http:' && protocol !== 'https:' && (protocol = 'https:');
 
-    // TDOO userinfo
+        const baseURL = `${protocol}//${host}${contextRoot || '/'}`;
+        let url = `${baseURL}config.js`;
 
-    const baseURL = `${protocol}//${host}${contextRoot || '/'}`;
-    let url = `${baseURL}config.js`;
+        // XXX In order to support multiple shards, tell the room to the deployment.
+        room && (url += `?room=${room.toLowerCase()}`);
 
-    // XXX In order to support multiple shards, tell the room to the deployment.
-    room && (url += `?room=${room.toLowerCase()}`);
+        let config;
 
-    /* eslint-enable no-param-reassign */
+        try {
+            config = await loadConfig(url);
+        } catch (error) {
+            config = restoreConfig(baseURL);
+            dispatch(loadConfigError(error, locationURL));
+        }
 
-    return loadConfig(url).then(
-        /* onFulfilled */ config => {
-            // FIXME If the config is no longer needed (in the terms of
-            // _loadConfig) and that happened because of an intervening
-            // _loadConfig for the same baseURL, then the unneeded config may be
-            // stored after the needed config. Anyway.
+        if (config) {
             dispatch(storeConfig(baseURL, config));
 
-            return config;
-        },
-        /* onRejected */ error => {
-            // XXX The (down)loading of config failed. Try to use the last
-            // successfully fetched for that deployment. It may not match the
-            // shard.
-            const config = restoreConfig(baseURL);
+            if (getState()['features/base/config'].locationURL === locationURL) {
+                dispatch(setLocationURL(locationURL));
+                dispatch(setConfig(config));
 
-            if (config) {
-                return config;
+                dispatch(setRoom(room));
+            } else {
+                dispatch(loadConfigError(new Error('Config no longer needed!'), locationURL));
```

Comment:
Does dispatching loadConfigError here again conflict with the dispatch of loadConfigError above?

Reply:
These are two different kind of failures. This one is when multiple "load config" tasks are started and it gets outdated - only the most recent one is to be accepted in the end. The other one above if it just fails to fetch.

Reply:
So there isn't a worry about the error here overriding the stored error above?

Reply:
Nope, the error will be stored only for the current request:
https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/config/reducer.js#L70